### PR TITLE
Drawer

### PR DESCRIPTION
**Pull request by ggrimaudDTS**

### CODE REVIEW

File: @utahdts/utah-design-system/react/components/containers/drawer/Drawer.jsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  const [lastActiveElement] = useImmer(/** @type {HTMLElement | undefined} */(document.activeElement));
+  const [firstTabElement, setFirstTabElement] = useImmer(/** @type {HTMLElement | undefined} */(undefined));
+  const [lastTabElement, setLastTabElement] = useImmer(/** @type {HTMLElement | undefined} */(undefined));
+  const { addAssertiveMessage } = useAriaMessaging();
+
+  const handleEscape = useHandleEscape(onEscape);
+  const handleTab = useHandleTab(firstTabElement, lastTabElement);
+
+  useEffect(() => {
+    if (ref) {
```
Comment (AustinHaws-Utah) [inline @utahdts/utah-design-system/react/components/containers/drawer/Drawer.jsx:45]: `ref` will always be "truthy". `ref.current` may be `null`. changing the `if` to be `if (ref.current)` would probably also remove the need for the `@ts-ignore`

File: @utahdts/utah-design-system/react/components/containers/drawer/Drawer.jsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+ * @param {DrawerPlacement} [props.position]
+ * @returns {JSX.Element}
+ */
+export function Drawer({
+  ariaLabelledBy,
+  children,
+  className,
+  id,
+  innerRef,
+  onEscape,
```
Comment (AustinHaws-Utah) [inline @utahdts/utah-design-system/react/components/containers/drawer/Drawer.jsx:31]: onEscape sorts after onClose... 🤷

File: @utahdts/utah-design-system/react/components/containers/drawer/DrawerContent.jsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { joinClassNames } from '../../../util/joinClassNames';
+
+/**
+ * @param {object} props
+ * @param {import('react').ReactNode} props.children
+ * @param {string} [props.className]
+ * @param {string} props.id
```
Comment (AustinHaws-Utah) [inline @utahdts/utah-design-system/react/components/containers/drawer/DrawerContent.jsx:7]: does `id` need to be required?

File: @utahdts/utah-design-system/react/components/containers/drawer/DrawerFooter.jsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { joinClassNames } from '../../../util/joinClassNames';
+
+/**
+ * @param {object} props
+ * @param {import('react').ReactNode} props.children
+ * @param {string} [props.className]
+ * @param {string} props.id
```
Comment (AustinHaws-Utah) [inline @utahdts/utah-design-system/react/components/containers/drawer/DrawerFooter.jsx:7]: does `id` need to be required?

File: @utahdts/utah-design-system/react/components/containers/drawer/DrawerTitle.jsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { joinClassNames } from '../../../util/joinClassNames';
+
+/**
+ * @param {object} props
+ * @param {import('react').ReactNode} props.children
+ * @param {string} [props.className]
+ * @param {string} props.id
```
Comment (AustinHaws-Utah) [inline @utahdts/utah-design-system/react/components/containers/drawer/DrawerTitle.jsx:7]: does `id` need to be required?

File: @utahdts/utah-design-system/react/components/popups/Modal/ModalTitle.jsx [hunk 6]
```diff
@@ -4,15 +4,18 @@ import { joinClassNames } from '../../../util/joinClassNames';
  * @param {object} props
  * @param {import('react').ReactNode} props.children
  * @param {string} [props.className]
+ * @param {string} props.id
```
Comment (AustinHaws-Utah) [inline @utahdts/utah-design-system/react/components/popups/Modal/ModalTitle.jsx:7]: does `id` need to be required?

File: @utahdts/utah-design-system/react/components/popups/Modal/ModalFooter.jsx [hunk 7]
```diff
@@ -4,15 +4,18 @@ import { joinClassNames } from '../../../util/joinClassNames';
  * @param {object} props
  * @param {import('react').ReactNode} props.children
  * @param {string} [props.className]
+ * @param {string} props.id
```
Comment (AustinHaws-Utah) [inline @utahdts/utah-design-system/react/components/popups/Modal/ModalFooter.jsx:7]: does `id` need to be required?

File: @utahdts/utah-design-system/react/hooks/useHandleEscape.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+ * @param {import('react').KeyboardEventHandler} [onEscape]
+ * @returns {(...args: any[]) => void} func
+ */
+export function useHandleEscape(onEscape) {
+  return useCallback(
+    (/** @type {import('react').KeyboardEvent<HTMLDialogElement>} */ e) => {
+      if (e.code === 'Escape' || e.key === 'Escape') {
+        e.preventDefault();
+        e.stopPropagation();
+        if (onEscape) { onEscape(e); }
```
Comment (AustinHaws-Utah) [inline @utahdts/utah-design-system/react/hooks/useHandleEscape.js:13]: sonar may ding this on having an if block on one line

File: @utahdts/utah-design-system/@types/jsDocTypes.d.js [hunk 9]
```diff
@@ -55,6 +55,11 @@
  * @typedef {'inline' | 'bottom' | 'bottom-left' | 'bottom-right' | 'top' | 'top-left' | 'top-right'} BannerPlacement
```
Comment (AustinHaws-Utah) [inline @utahdts/utah-design-system/@types/jsDocTypes.d.js:55]: General note: this appears to have been branched from the color picker stuff? So it has all the icons and color picker work in it that is not yet in the 2024Q3S5 branch. Maybe review those first before merging this?

File: @utahdts/utah-design-system/react/components/containers/drawer/Drawer.jsx (hunk 2 above)
Comment (ggrimaudDTS) [reply @utahdts/utah-design-system/react/components/containers/drawer/Drawer.jsx:31]: 😨

File: @utahdts/utah-design-system/react/components/containers/drawer/DrawerContent.jsx (hunk 3 above)
Comment (jmsharp-utah) [reply @utahdts/utah-design-system/react/components/containers/drawer/DrawerContent.jsx:7]: `id` is not required

File: @utahdts/utah-design-system/react/components/containers/drawer/DrawerFooter.jsx (hunk 4 above)
Comment (jmsharp-utah) [reply @utahdts/utah-design-system/react/components/containers/drawer/DrawerFooter.jsx:7]: `id` is not required

File: @utahdts/utah-design-system/react/components/containers/drawer/DrawerTitle.jsx (hunk 5 above)
Comment (jmsharp-utah) [reply @utahdts/utah-design-system/react/components/containers/drawer/DrawerTitle.jsx:7]: Put a comment on the jsdoc to explain its use?

File: @utahdts/utah-design-system/react/components/popups/Modal/ModalFooter.jsx (hunk 7 above)
Comment (jmsharp-utah) [reply @utahdts/utah-design-system/react/components/popups/Modal/ModalFooter.jsx:7]: `id` is not required

File: @utahdts/utah-design-system/react/components/popups/Modal/ModalTitle.jsx (hunk 6 above)
Comment (jmsharp-utah) [reply @utahdts/utah-design-system/react/components/popups/Modal/ModalTitle.jsx:7]: Maybe add a comment in the jsdoc to explain it

File: utah-design-system-website/src/react/components/websiteContent/library/components/containers/Drawer/examples/SimpleDrawer.jsx [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+              Nunc rhoncus ligula in tellus venenatis pulvinar. Class aptent taciti sociosqu ad litora
+              torquent per conubia nostra, per inceptos himenaeos.
+            </DrawerContent>
+            <DrawerFooter className="flex justify-end" id="simple-modal-id">
+              <Button onClick={closeDrawer} className="mr-spacing">Cancel</Button>
+              <Button onClick={closeDrawer} className="button--solid button--primary-color">Okay</Button>
+            </DrawerFooter>
+          </Drawer>
+        )
+        : ''}
```
Comment (jmsharp-utah) [inline utah-design-system-website/src/react/components/websiteContent/library/components/containers/Drawer/examples/SimpleDrawer.jsx:56]: Should be a `null`? -Austin

**Review by jmsharp-utah — APPROVED** [review]
🗄️ ---> 🔪 🩸 💀  ---> 🪦 😢

**Pull request merged**